app/utils/geostore.py

```diff
@@ -51,3 +73,54 @@ async def get_geostore(geostore_id: UUID, geostore_origin: str) -> GeostoreCommo
         raise HTTPException(status_code=500, detail=str(e))
     except BadResponseError as e:
         raise HTTPException(status_code=400, detail=str(e))
+
+
+async def get_geostore_from_any_origin(
+    geostore_id: UUID, geostore_origin: GeostoreOrigin
+) -> GeostoreCommon:
+    """Looks for geometry in all geostores, beginning with client's choice."""
+
+    geostore_constructor = {
+        GeostoreOrigin.gfw: _get_gfw_geostore,
+        GeostoreOrigin.rw: rw_api.get_geostore,
+    }
+
+    geo_func = geostore_constructor.pop(geostore_origin)
+
+    # If we get a geostore from any origin return it,
+    # if we get all 404s return a 404,
+    # if we get a mixture of errors return a 500 with explanation
+    exceptions: List[Exception] = []
+    try:
+        return await geo_func(geostore_id)
+    except RecordNotFoundError as e:
+        exceptions.append(e)
+    except Exception as e:
+        logger.exception(e)
+        exceptions.append(e)
+
+    # Will we really ever have >2 geostore sources?
+    # Preserve the possibility for now.
+    for geo_func in geostore_constructor.values():
+        try:
+            return await geo_func(geostore_id)
+        except RecordNotFoundError as e:
+            exceptions.append(e)
+        except Exception as e:
+            logger.exception(e)
+            exceptions.append(e)
+
+    non_404s = [
+        exception
+        for exception in exceptions
+        if not isinstance(exception, RecordNotFoundError)
+    ]
+
+    if non_404s:
+        msg = (
+            f"One or more errors were encountered looking for geostore "
+            f"{geostore_id}. Please email info@wri.org for help."
+        )
```

Comment:
I like this. So, who monitors `info@wri.org`?

Reply:
A good question. I am no longer sure this is the best address to use. data@wri.org would probably be better, I think that's monitored by Logan. But I'll ask him. I thought there was one monitored by Maanas, who is perhaps the best. For the moment I'll set it to mine, as it's the only gentlemanly thing to do in doubt.

Reply:
Ah, okay. Yeah, we should definitely find the _official_ email for this.